projects/components/src/table/renderers/data-cell/table-data-cell-renderer.component.ts

```diff
@@ -1,54 +1,95 @@
-import { ChangeDetectionStrategy, Component, Injector, Input, OnInit } from '@angular/core';
-import { Dictionary } from '@hypertrace/common';
-import { TableColumnConfig } from '../../table-api';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ComponentFactoryResolver,
+  ComponentRef,
+  Injector,
+  Input,
+  OnInit
+} from '@angular/core';
+import { FilterAttribute } from '../../../filter-bar/filter-attribute';
+import { TableColumnConfig, TableRow } from '../../table-api';
 import { TableCellAlignmentType } from '../table-cell-alignment-type';
 import { TableCellRendererConstructor } from '../table-cell-renderer';
+import { TableCellRendererComponent } from '../table-cell-renderer.component';
 import { TableCellRendererService } from '../table-cell-renderer.service';
 
 @Component({
   selector: 'htc-table-data-cell-renderer',
   styleUrls: ['./table-data-cell-renderer.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <div class="vertical-alignment-container">
-      <div class="horizontal-alignment-container" [ngClass]="this.alignment">
+    <div class="table-data-cell-renderer" [class.selected]="this.popoverOpen">
+      <htc-filter-button
+        class="filter-button"
+        *ngIf="this.columnConfig?.filterable && this.leftAlignFilterButton"
+        [metadata]="this.metadata"
+        [attribute]="this.columnConfig?.attribute"
+        [value]="this.componentRef?.instance.value"
+        (popoverOpen)="this.popoverOpen = $event"
+      ></htc-filter-button>
```

Comment:
there's enough in this that it might be worth using a template rather than repeating it for either alignment

Reply:
Fair. Let's see how all of the other fixes go and I can circle back to this if I have time.